src/textual/_segment_tools.py

```diff
@@ -129,19 +129,39 @@ def line_trim(segments: list[Segment], start: bool, end: bool) -> list[Segment]:
         A new list of segments.
     """
     segments = segments.copy()
-    if segments and start:
-        _, first_segment = segments[0].split_cells(1)
-        if first_segment.text:
-            segments[0] = first_segment
-        else:
+
+    # Empty segments should be ignored as they don't contribute to cell width
+    while segments and start:
```

Comment:
Now that the root cause of empty segments is fixed, I'd be tempted to roll this back. We should treat "no empty segments" as an invariant.

Since it is easy to do accidentally, maybe we should put in a check wrapped with `if app.debug` that logs a warning if there are empty segments.

Reply:
Do you think we should put this check in this function or should it be an invariant in broader Textual that "empty, non-control Segments don't make sense"?

I've found a bunch of places that I think could produce empty Segments if we're not careful, so if you think that there's lots of code that wouldn't handle that correctly then maybe it should be a broader check?

Reply:
Yeah, not here. Perhaps `_styles_cache.py`.

Reply:
Ok, I've added something to the styles cache, let me know what you think